### PR TITLE
Feat/discovery page UI

### LINKED ISSUE #260: Implement UI of discovery page
We have to implement a new discovery page allowing users to explore different trips that were shared by other users or their friends, this task consists of implementing the UI of this page before deciding how to share trips

### PR DESCRIPTION
**Pull request by apana03**
## Summary

This PR aims to introduce a new feature to the app. The Discover page where users can see trips that are public within the app in order to get ideas from other people.

## Changes

- **Models:** Added the getFeed function in the TripViewModel.
- **Screens/UI:** Updated the Search screen to include the new Discover tab, created a new composable the DiscoverContent.
- **Logic/Controllers:**  Updated the Navigation actions class to take into account this new tab.

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #260.
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached (if applicable).
- [x] Documentation has been updated (if applicable).

##  Testing
- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

Tested manually on emulator also added plenty of other unit tests in different classes.

##  Related Issues/Tickets

Closes #260.
Opens #277.

## Screenshots (if applicable)

https://github.com/user-attachments/assets/641676cc-0fb0-4e53-a266-d144b4bcfe51

* Note that the space in the trips seen in the discover tab comes from the fact that those give screens do not have any descriptions.

### CODE REVIEW

File: app/src/test/java/com/android/voyageur/model/trip/TripsViewModelTest.kt [hunk 1]
```diff
@@ -335,4 +335,21 @@ class TripsViewModelTest {
...
     assert(tripsViewModel.uiState.value is UiState.Initial)
   }
+
+  @Test
+  fun testGetFeed() {
+    tripsViewModel.getFeed("userId")
+    verify(tripsRepository).getFeed(any(), any(), any())
+  }
+
+  @Test
```
Comment (mirceagavrizi) [inline app/src/test/java/com/android/voyageur/model/trip/TripsViewModelTest.kt:345]: Can you please also add tests for these 2 methods in TripsViewModelTest?

Comment (apana03) [reply app/src/test/java/com/android/voyageur/model/trip/TripsViewModelTest.kt:345]: I do not really get you comment the getFeed function I added in the TripViewModel is tested in this PR. What 2 methods are you referring to?

File: app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    painter = rememberAsyncImagePainter(trip.imageUri),
+                    contentDescription = "Trip Image",
+                    contentScale = ContentScale.Crop,
+                    modifier =
+                        Modifier.fillMaxWidth().height(200.dp).clip(RoundedCornerShape(8.dp)))
+              } else {
+                Image(
+                    painter = rememberAsyncImagePainter(model = R.drawable.default_trip_image),
+                    contentDescription = "Trip Image",
+                    contentScale = ContentScale.Crop,
```
Comment (mirceagavrizi) [inline app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt:85]: For the contentDescription in the if else structure you can consider naming them with something like: "Image of ${trip.name}", for better accuracy. Also, I recommend adding testTags to all the components in order to facilitate the testing.

**Review by mirceagavrizi — COMMENTED** [review]
On a more general note, what should be the expected behavior of the search bar in the discover trip screen? Let me know your ideas.
Moreover, please make sure to reach 80% line coverage for the new code or let me know if you plan on making another PR for this.

**Comment by apana03** [comment]
> On a more general note, what should be the expected behavior of the search bar in the discover trip screen? Let me know your ideas. Moreover, please make sure to reach 80% line coverage for the new code or let me know if you plan on making another PR for this.

The search bar has no purpose when in the Discover tab, it has a purpose in the other 2 tabs, but since it is part of the search screen and the tabs only modify the content of that screen I should not remove. However if you think it should also filter the results in the discover screen let me know and I can add this functionality in a future PR. I tried to replicate the behaviour of the reel section from Instagram that is why.

Also Managed to get the line coverage to 93%.

**Review by mirceagavrizi — APPROVED** [review]
Thank you for taking into consideration my suggestions! Great job implementing this screen and we will see at the meeting what exactly will be displayed on the read-only card.

**Pull request merged**